crates/daphne/src/roles/helper/handle_agg_job.rs

```diff
@@ -0,0 +1,321 @@
+    request: DapRequest<AggregationJobInitReq>,
+}
+
+/// The transition between [`WithTaskConfig`] and [`InitializedReports`], this allows for the
+/// costumization of how reports get initialized.
+///
+/// This type is returned by [`HandleAggJob::into_parts`] and [`Self::with_initialized_reports`]
+/// can be used to return to the [`HandleAggJob`] state machine flow.
+#[non_exhaustive]
+pub struct ToInitializedReportsTransition {
+    pub task_id: TaskId,
+    pub part_batch_sel: PartialBatchSelector,
+    pub task_config: DapTaskConfig,
+}
+
+/// The reports have been initialized and are ready for aggregation.
+pub struct InitializedReports {
+    task_id: TaskId,
+    part_batch_sel: PartialBatchSelector,
+    task_config: DapTaskConfig,
+    reports: Vec<InitializedReport<WithPeerPrepShare>>,
+}
+
+macro_rules! impl_from {
+    ($($t:ty),*$(,)?) => {
+        $(impl From<$t> for HandleAggJob<$t> {
+            fn from(state: $t) -> Self {
+                Self { state }
+            }
+        })*
+    };
+}
+
+impl_from!(Init, WithTaskConfig, InitializedReports);
+
+pub fn start(request: DapRequest<AggregationJobInitReq>) -> HandleAggJob<Init> {
+    HandleAggJob::new(request)
+}
+
+impl HandleAggJob<Init> {
+    pub fn new(request: DapRequest<AggregationJobInitReq>) -> Self {
+        Self {
+            state: Init(request),
+        }
+    }
+
+    /// Resolve the task config in the default way.
+    pub async fn resolve_task_config<A: DapHelper>(
+        self,
+        aggregator: &A,
+    ) -> Result<HandleAggJob<WithTaskConfig>, DapError> {
+        let task_config = resolve_task_config(aggregator, &self.state.0.meta).await?;
+        self.with_task_config(task_config)
+    }
+
+    /// Provide the resolved task configuration.
+    pub fn with_task_config(
+        self,
+        task_config: DapTaskConfig,
+    ) -> Result<HandleAggJob<WithTaskConfig>, DapError> {
+        let Self {
+            state: Init(request),
+        } = self;
+
+        check_part_batch(
+            &request.task_id,
+            &task_config,
+            &request.payload.part_batch_sel,
+            &request.payload.agg_param,
+        )?;
+        Ok(HandleAggJob {
+            state: WithTaskConfig {
+                task_config,
+                request,
+            },
+        })
+    }
+}
+
+impl HandleAggJob<WithTaskConfig> {
+    /// Initialize the reports in the default way.
+    pub async fn initialize_reports<A: DapHelper>(
+        self,
+        aggregator: &A,
+        replay_protection: crate::ReplayProtection,
+    ) -> Result<HandleAggJob<InitializedReports>, DapError> {
+        let WithTaskConfig {
+            task_config,
+            request,
+        } = self.state;
+        let task_id = request.task_id;
+        let part_batch_sel = request.payload.part_batch_sel.clone();
+        let initialized_reports = task_config.consume_agg_job_req(
+            &aggregator.get_receiver_configs(task_config.version).await?,
```

Comment:
This would make the code a little cleaer
```suggestion
            &aggregator.get_hpke_receiver_configs(task_config.version).await?,
```